### DISCUSSION TITLE
How to provide initial estimates

### DISCUSSION THREAD

**Discussion by harishkaushikbugworks** [nlmixrdevelopment/nlmixr#496] [Q&A]
Hello Matt,
I am fitting my oral data to 1 and 2 cmt models using nlmixr, i estimated the parameters using WinNonlin as follows

the data (conc in ng/mL), do i have to change it to ug/mL 
for ex: Vc=360248mL and CL67878mL/hr, should i convert to L and L/hr 
should i provide the initiatal estimate as such ie 67878 mL/hr or 67.878L/hr or Log (parameter value)

it is very confusing, can you help me

Iam attaching the raw data and code i tried for fitting

Regards
Harish
[nlmixr Code.txt](https://github.com/nlmixrdevelopment/nlmixr/files/6285135/nlmixr.Code.txt)

[200mgQD.xlsx](https://github.com/nlmixrdevelopment/nlmixr/files/6285143/200mgQD.xlsx)

**Comment by mattfidler** [comment]
You need to figure out the amount units that you have in your dataset.  Then your concentrations are:

`amt units / volume units`

If your actual concentrations are in a different sort of unit, you need to convert them to match the concentrations in your dataset.  

The scaling factor is easy to calculate by hand, but you can also calculate it in R using `units`.  Assuming your dose is in `mg` and you are estimating volume in terms of `mL` you would have:

``` r
library(units)
#> udunits system database from C:/R/r-4.0.3/library/units/share/udunits
library(tidyverse)

set_units(1, mg/mL) %>% set_units(ng/mL)
#> 1e+06 [ng/mL]
```

<sup>Created on 2021-04-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>


You could then write

```r
cp = amt/v*1e6 # concentration in ng/mL
```

**Comment by harishkaushikbugworks** [comment]
Thanks Matt,

Question is on initial estimates should I directly plug values as such (meaning parameters estimated from mean plot, using winnonlin) should I apply log () values

Regards
Harish
________________________________

**Comment by mattfidler** [comment]
If you are estimating values like:

```r
 cl <- exp(tcl + eta.cl)
```
Then you would provide estimates on a log scale.  You can do this directly in the ini section
```r
ini({
cl <- log(30)
})
```

**Comment by mattfidler** [comment]
If you are estimating values like:
```r
 cl <- tcl + eta.cl
```

You would provide estimates on a linear scale:

```r
cl <- 30
```

**Comment by harishkaushikbugworks** [comment]
Thanks Matt,
Sorry for dragging this discussion, in the example you provided can I assume Cl value of 30 is estimated outside using some way either by WNL or literature values, then directly take that value as log(30) directly in nlmixr, is it applicable only for CL or V parameters also, I have also seen few case studies where log of parameters value is not used, so when we have to use such kind of initial estimates

Please correct if my understanding is wrong

Regards
Harish
________________________________

**Comment by mattfidler** [comment]
You can use the `log` estimate whenever it is wrapped in `exp(tx)` this applies to `V` and any other parameter.  You can also use `logit()` and `expit()` for  values you mean to constrain in your estimation
